### ISSUE TITLE
allowUnusedForeachVariables should be for values only

### ISSUE THREAD

**Issue by kkmuffme** [sirbrillig/phpcs-variable-analysis#166]
allowUnusedForeachVariables should be for $values only, unused $keys should ALWAYS get reported

Since unused $keys never make sense, but unused $values may be a necessity when you want to use the keys only
(which is why I suggested the rule originally)

**Comment by sirbrillig** [comment]
> Since unused $keys never make sense,

That's an interesting point. I had forgotten that `foreach` on an associative array without the `key => value` syntax will iterate over values, and I must not be the only one because I've seen a lot of code of the form:

```php
foreach ($foo as $key => $value) {
  doSomethingWith($value);
}
```

So this is a good point, and perhaps the current behavior is actually a bug. I'll make a patch to fix this.
